File: javascript/grade-school/grade-school.js
//
// This is only a SKELETON file for the 'Grade School' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class GradeSchool {
  constructor() {
    this.db = {};
  }

  roster() {
    let clone = { ...this.db };
    const rost = Object.keys(clone).reduce((acc, key) => {
      acc[key] = clone[key].sort();
      return acc;
    }, {});
    return JSON.parse(JSON.stringify(rost));
  }

  add(name, grade) {
    if (!this.db.hasOwnProperty(grade)) {
      this.db[grade] = [];
    }
    this.db[grade].push(name);
  }

  grade(grade) {
    let clone = { ...this.db };
    const gradeRoster = clone[grade] ? clone[grade].sort() : [];
    return JSON.parse(JSON.stringify(gradeRoster));
  }
}

File: python/grade-school/grade_school.py


class School(object):
    def __init__(self):
        self.grades =   {
                "1": [],
                "2": [],
                "3": [],
                "4": [],
                "5": [],
                "6": [],
                "7": [],
                "8": [],
                "9": [],
                "10": [],
                "11": [],
                "12": []
            }

    def add_student(self, name, grade):
        grd = str(grade)
        self.grades[grd].append(name)

    def roster(self):
        roster = [student for i in range(1, 13) for student in sorted(self.grades[str(i)])]
        return roster

    def grade(self, grade_number):
        return sorted(self.grades[str(grade_number)])

